sstables/sstables.hh

```diff
@@ -498,6 +498,7 @@ public:
         virtual future<> wipe(const sstable& sst) noexcept = 0;
         virtual future<file> open_component(const sstable& sst, component_type type, open_flags flags, file_open_options options, bool check_integrity) = 0;
         virtual future<data_sink> make_component_sink(sstable& sst, component_type type, io_priority_class pc) = 0;
+        virtual future<data_sink> make_component_sink(sstable& sst, component_type type, open_flags oflags, file_output_stream_options options) = 0;
```

Comment:
These two functions aren't identical, so they should have different names. Overloading a name should be done only if the operation is the same.